sdk/storage/azure_storage_blob/src/models/blob_properties.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::headers::Headers;
+
+/// Properties of an Azure Storage blob.
+///
+#[derive(Clone, Default, Debug)]
+pub struct BlobProperties {
+    pub access_tier_inferred: String,
+    pub access_tier: String,
+    pub blob_type: String,
+    pub content_length: String,
+    pub content_md5: String,
+    pub content_type: String,
+    pub creation_time: String,
+    pub etag: String,
+    pub last_modified: String,
+    pub lease_state: String,
+    pub lease_status: String,
+    pub server_encrypted: String,
+}
+
+impl BlobProperties {
+    pub fn build_from_response_headers(response_headers: &Headers) -> BlobProperties {
```

Comment:
Does this really need to be public? Better to keep things private (or `pub(crate)` until we know they need to be public.

Reply:
I also recommend implementing `AsHeaders` and `FromHeaders` instead, which plugs into our system better. see `typespec_client_core::http::headers` and you can find examples therein as well.

Reply:
Many of these should also be typed headers, which also corresponds to their fields being defined as those types on your struct. `Etag` is a good example.

Whatever is typed in Go should be typed here as well.

Reply:
That lets you call `let properties: BlobProperties = headers().get()?;` for example.

Reply:
Hi @heaths Heath, in regards to all the feedback regarding our handwritten `BlobProperties` model-- I can work on getting this implementation to better, but this was mostly a stopgap implementation since it is still a pending discussion if these sort of models will be auto-generated and available in time for the MVP. I will leave this thread open and can revisit this later on- but I don't want to this to be blocking unless we are going to ship with our handwritten models.